cargo-near/src/build.rs

```diff
@@ -18,31 +17,27 @@ pub(crate) fn run(args: BuildCommand) -> anyhow::Result<()> {
         args.manifest_path.unwrap_or_else(|| "Cargo.toml".into()),
     )?)?;
 
-    let out_dir = args
-        .out_dir
-        .unwrap_or_else(|| crate_metadata.target_directory.clone());
-    fs::create_dir_all(&out_dir)
-        .with_context(|| format!("failed to create directory `{}`", out_dir.display()))?;
-    let out_dir = out_dir
-        .canonicalize()
-        .with_context(|| format!("failed to access output directory `{}`", out_dir.display()))?;
+    let out_dir = util::force_canonicalize_dir(
+        &args
+            .out_dir
+            .unwrap_or_else(|| crate_metadata.target_directory.clone()),
+    )?;
 
     let mut build_env = vec![("RUSTFLAGS", "-C link-arg=-s")];
     let mut cargo_args = vec!["--target", COMPILATION_TARGET];
     if args.release {
         cargo_args.push("--release");
     }
 
-    let abi = (!args.no_abi)
-        .then(|| abi::write_to_file(&crate_metadata, args.doc))
-        .transpose()?;
+    let mut abi_path = None;
+    if !args.no_abi {
+        let AbiResult { path } = abi::write_to_file(&crate_metadata, args.doc)?;
+        abi_path.replace(util::copy(&path, &out_dir)?);
+    }
 
-    let wasm_artifact = if args.embed_abi {
+    let mut wasm_artifact = if let Some(ref abi_path) = abi_path {
```

Comment:
Wait, did you intend to get rid of the `args.embed_abi` check?

Reply:
Fixed.